### ISSUE TITLE
Clarify Active Directory Requirements

### ISSUE THREAD

**Issue by Mike-Crowley** [MicrosoftDocs/windowsserverdocs#7345]
In "Step 2: Configure SMB over QUIC \ #2" it says join the server to the domain, however later in the article (twice), it is implied you can host SMB over QUIC from a server on a workgroup. 

It is also not entirely clear if the client computers need to be joined to a domain, and if so, why. Can you clarify this? Customers with AADJ-workstations may want to use this solution for workgroup servers tied to Azure storage.

Clarification is also required if it is acceptable to use Azure AD Domain services on the server, while using AADJ on the workstations.  For example, can you write SPNs to AAD DS?

A section discussing the overall AD requirements / supportability matrix would be useful.

---
#### Document Details

⚠ *Do not edit this section. It is required for learn.microsoft.com ➟ GitHub issue linking.*

* ID: 64dadae1-7656-54e0-099d-620229836135
* Version Independent ID: 8f86a34d-ce81-d571-e118-29169a2c0df3
* Content: [SMB over QUIC](https://learn.microsoft.com/en-us/windows-server/storage/file-server/smb-over-quic#configure-the-kdc-proxy-optional-but-recommended)
* Content Source: [WindowsServerDocs/storage/file-server/smb-over-quic.md](https://github.com/MicrosoftDocs/windowsserverdocs/blob/main/WindowsServerDocs/storage/file-server/smb-over-quic.md)
* Product: **windows-server**
* Technology: **storage**
* GitHub Login: @nedpyle
* Microsoft Alias: **inhenkel**

**Comment by nedpyle** [comment]
It’s talking about the 98% use case, which would be domain-joined. You cannot join Windows Server to AAD, so they would purely be in a workgroup and using NTLM local user creds from the clients; AAD doesn’t provide Windows remote authentication. SMB over QUIC doesn’t require AD.

I will update this doc to clarify it, this is good feedback.

NP | PM

From: Mike Crowley ***@***.***>
Sent: Wednesday, June 21, 2023 2:01 PM
To: MicrosoftDocs/windowsserverdocs ***@***.***>
Cc: Ned Pyle ***@***.***>; Mention ***@***.***>
Subject: [MicrosoftDocs/windowsserverdocs] Clarify Active Directory Requirements (Issue #7345)


In "Step 2: Configure SMB over QUIC \ #2<https://github.com/MicrosoftDocs/windowsserverdocs/pull/2>" it says join the server to the domain, however later in the article (twice), it is implied you can host SMB over QUIC from a server on a workgroup.

It is also not entirely clear if the client computers need to be joined to a domain, and if so, why. Can you clarify this? Customers with AADJ-workstations may want to use this solution for workgroup servers tied to Azure storage.

Clarification is also required if it is acceptable to use Azure AD Domain services on the server, while using AADJ on the workstations. For example, can you write SPNs to AAD DS?

A section discussing the overall AD requirements / supportability matrix would be useful.

________________________________
Document Details

⚠ Do not edit this section. It is required for learn.microsoft.com ➟ GitHub issue linking.

  *   ID: 64dadae1-7656-54e0-099d-620229836135
  *   Version Independent ID: 8f86a34d-ce81-d571-e118-29169a2c0df3
  *   Content: SMB over QUIC<https://learn.microsoft.com/en-us/windows-server/storage/file-server/smb-over-quic#configure-the-kdc-proxy-optional-but-recommended>
  *   Content Source: WindowsServerDocs/storage/file-server/smb-over-quic.md<https://github.com/MicrosoftDocs/windowsserverdocs/blob/main/WindowsServerDocs/storage/file-server/smb-over-quic.md>
  *   Product: windows-server
  *   Technology: storage
  *   GitHub Login: @nedpyle<https://github.com/nedpyle>
  *   Microsoft Alias: inhenkel

**Comment by Mike-Crowley** [comment]
Hey Ned, you're everywhere!

You actually can join WS 2022 to AAD. If you say 98%, sure, but I see this as a big win for people trying to get rid of legacy stuff, but can't fit into onedrive/spo - so in that case, workstations would be AADJ, and minimal servers would be joined to AAD DS or AAD directly, such as maybe in this case.

**Comment by Mike-Crowley** [comment]
E.g, I have customers who are using M365 where possible, and they have joined their Windows 11 machines directly to AAD. There is still a "legacy" AD DS, which feeds hybrid identities to Azure, for things like Azure Kerberos, however we don't really want to deploy new things to the legacy AD DS.

We have deployed AAD DS to support some crappy apps, which run on Windows and joined to AAD DS. 

We also have some servers directly joined to AAD (only supported in Windows 2022 Azure Edition), and this is the cleanest option IMO.

So ideally, we'd have the workstations talk do the Windows 2022 AADJ server using QUIC. We could use a commercial cert on the host for 443 and workstations could have a mapping script pushed via intune.

**Comment by nedpyle** [comment]
I am indeed everywhere!

Sorry, I meant you cannot natively do so – The AADJ team does not support or provide management code in Windows Server for AADJ, we’re actually working on building that right now. That it works in Azure IaaS only is through some trickery they built, and that won’t work if you were to run Windows Server on Azure Stack HCI, for instance. It’s not specific to Azure Edition, it’s specific to Azure IaaS and a special agent they wrote just for themselves. I can say this with authority because I am the creator of Azure Edition 😊 and involved in some of the AADJ work becoming native for the past 3 years (you can see how fast that’s going!)

It doesn’t matter though because you cannot use an AAD cred to remotely access an SMB share (or any Windows feature remotely – event log, remote registry, etc.) even if AAD joined – there is no user SID. You will always have to use a local user with NTLM if you’re not classic AD domain joined.

I am going to make this article less sucky nonetheless, it’s open on my desk right now!

NP | PM

**Comment by Mike-Crowley** [comment]
I dont doubt there is trickery involved, but if you open sysdm.cpl it shows its joined to Azure AD and also it explicitly says so in the documentation: 

["After you enable this capability, your Windows VMs in Azure will be Azure AD joined."](https://learn.microsoft.com/en-us/azure/active-directory/devices/howto-vm-sign-in-azure-ad-windows)

**Comment by Mike-Crowley** [comment]
>It doesn’t matter though because you cannot use an AAD cred to remotely access an SMB share (or any Windows feature remotely – event log, remote registry, etc.) even if AAD joined – there is no user SID. You will always have to use a local user with NTLM if you’re not classic AD domain joined.

You can if its AAD DS joined, which is the 3rd option. in this case you can use your AAD cred.

**Comment by nedpyle** [comment]
Yes but try to make it work anywhere else outside Azure IaaS, I dare you 😊.  I have a team of half a dozen people working on this today.

I want to be really clear here though: joining AAD anywhere, in Windows client, in Azure IaaS VMs, etc does not let you access Windows remote resources. There is no AAD package in SPNEGO, no auth token, no security mechanism in Windows. AADJ only acts like a secure channel trust, it can’t access an SMB share or anything as a user. Any use of AAD to access something would require torque conversion elsewhere first.

NP | PM

**Comment by nedpyle** [comment]
You are not truly using AAD in that case - you are still using Active Directory for the auth in Windows. But I wasn't aware that they were doing something here to let you start with an AAD cred. Can you point me to the docs that talk about this? I'll be happy to add that as a possible option in the docs once I understand it better.

Do you actually have people using AADDS??? That thing is... extremely not popular.

NP | PM

**Comment by Mike-Crowley** [comment]
Thanks Ned, I'll get back to you in the AM. I apologize for the scope creep of this doc bug.

**Comment by gti-masonw** [comment]
I have an AAD tenant and cloud-only AAD users, lets say domain.com.  I setup Entra ID Managed Domain services - lets say aadds.domain.com. I have reset AAD user's passwords so they synced down to the managed domain, aadds.domain.com. I have setup SMB over QUIC and KDC proxy. I'm able to auto map SMB shares with AAD account credentials via intune. That works great, and is neato. However, I'm having trouble understanding how to get Windows Hello Cloud Trust enabled on an Entra managed domain. I have the UseCloudTrustForOnPremAuth policy enabled on my windows 11 clients. However, I'm hitting a road block with the AzureADHybridAuthenticationManagement module and can't use Set-AzureADKerberosServer to set the "on-prem" krb users and resources because I lack domain admin permissions with a entra managed domain. I realize this is a fairly bizarre use case, and maybe I'm missing something simple. Any ideas on how to create the kerberos resources in a managed domain? I'm so close - and hoping I don't need to go backwards and setup a true ADDS environment.

**Comment by robinharwood** [comment]
We hugely appreciate your contribution, feedback, and support to our technical documentation.

We’ve phased out GitHub Issues on this repository as the feedback mechanism for content and have replaced it with a new feedback system. For more information see, [Content User Feedback](https://aka.ms/ContentUserFeedback). As part of this rollout, we are migrating your issue where our team will use your comments to improve the documentation experience. Unfortunately, this does mean you won't be able to access this issue after the migration is complete.

All issues will be migrated by Thursday 29th August 2024.

Once again, thank you for your valuable feedback, we hope you’ll continue to contribute using our new feedback system.
